openmdao/solvers/nonlinear/newton.py

```diff
@@ -100,6 +100,11 @@ def _setup_solvers(self, system, depth):
         """
         super(NewtonSolver, self)._setup_solvers(system, depth)
 
+        # check for any discrete variables
+        if system._var_allprocs_discrete['output']:
+            raise RuntimeError("System '%s' has a Newton solver and contains discrete outputs %s." %
```

Comment:
I think newton could work on a system that contains discrete outputs, but only if that output doesn't connect to an input of a component that is in that system. (i.e., it could be pushing a discrete variable to a component in a higher group after this group converges, which would be fine conceptually.)

It would also work for all cases if the sending and receiving component are both under an 'fd' group under the Newton's group.

Reply:
Also, anything we do to Newton also applies to Broyden, our other solver that uses derivatives.

Reply:
Jubs says *any* discrete output in the system being solved by Newton/Broyden is invalid, and being under an fd group doesn't make it valid either.

Reply:
I added a check and a test for broyden.